Zend/zend_constants.c

```diff
@@ -456,7 +456,7 @@ ZEND_API zval *zend_get_constant_ex(zend_string *cname, zend_class_entry *scope,
 
 	if (!c) {
 		if (!(flags & ZEND_FETCH_CLASS_SILENT)) {
-			zend_throw_error(NULL, "Undefined constant '%s'", name);
+			zend_throw_error(NULL, "Undefined constant %s", name);
```

Comment:
And what do you think about the `Constant %s is deprecated` error? Should I use quotes here as well? Besides, we have the `Undefined constant %s::%s` message where I'm unsure what to do.

I can come up with two possible logic:
- We only quote global constants, and class constants remain unquoted
- We treat the undefined constant error messages specially because they display non-existent (possibly bogus) constant names?

Reply:
My recent commit quotes many non-existent or unfound symbols (constants, functions, methods, properties). I think I like this format, because in most cases these symbol names can be empty, or contain bogus characters, if I got it right. How do like the new format?

However, the recently changed `Undefined variable $%s` error message is an exception to the others, because I didn't quote it yet. Do you think it's a good idea to quote the variable name?

Reply:
I think the new version went a bit too far in the other direction. The main problem with the `Undefined constant %s` error is that it becomes just "Undefined constant" if the name is empty, which has no indication that the name is actually empty (it would make sense as a standalone message).

I think the right rule of thumb here is to quote things that appear standalone, such as constant names, but not things that appear together with other symbols, such as undefined variables, methods etc. In particular, I don't think that `Call to undefined method "Foo::()"` is any clearer than `Call to undefined method Foo::()`.

I guess another possibility, that would work for most things apart from constant names, is to check whether the symbol name contains invalid characters, and in that case print `Call to undefined method Foo::{""}()` and `Undefined variable: ${"\xff"}` and similar. Not sure if worth it though.

Reply:
Thank you for the suggestions, I wasn't able to come up with so sensible rule of thumbs :) I'll try to get rid of unnecessary quotes in my next commit.